v2/datastream-common/src/main/java/com/google/cloud/teleport/v2/datastream/transforms/FormatDatastreamRecordToJson.java

```diff
@@ -494,14 +495,14 @@ static void handleLogicalFieldType(
         Duration duration = Duration.ofMillis(((Long) element.get(fieldName)));
         jsonObject.put(fieldName, duration.toString());
       } else if (fieldSchema.getLogicalType() instanceof LogicalTypes.TimestampMicros) {
-        Long nanoseconds = (Long) element.get(fieldName) * TimeUnit.MICROSECONDS.toNanos(1);
-        Instant timestamp =
-            Instant.ofEpochSecond(
-                TimeUnit.NANOSECONDS.toSeconds(nanoseconds),
-                nanoseconds % TimeUnit.SECONDS.toNanos(1));
-        jsonObject.put(
-            fieldName,
-            timestamp.atOffset(ZoneOffset.UTC).format(DEFAULT_TIMESTAMP_WITH_TZ_FORMATTER));
```

Comment:
Thanks for the change.
Could we retain writing `Instant` to the Json object if it's not causing a loss of precision?

Feel free to close this if integration tests on spanner pass. 


Ref - Even before [cl/598730704](https://critique.corp.google.com/cl/598730704) where the typecasting but got introduced json was populated by `Instant`

Reply:
Thanks for fixing this.